server/src/main/java/org/opensearch/index/search/comparators/HalfFloatComparator.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.search.comparators;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.sandbox.document.HalfFloatPoint;
+import org.apache.lucene.search.LeafFieldComparator;
+import org.apache.lucene.search.comparators.NumericComparator;
+
+import java.io.IOException;
+
+/**
+ * The comparator for half_float numeric type.
+ * Comparator based on {@link Float#compare} for {@code numHits}. This comparator provides a
+ * skipping functionality – an iterator that can skip over non-competitive documents.
+ */
+public class HalfFloatComparator extends NumericComparator<Float> {
+    private final float[] values;
+    protected float topValue;
+    protected float bottom;
+
+    public HalfFloatComparator(int numHits, String field, Float missingValue, boolean reverse, boolean enableSkipping) {
+        super(field, missingValue != null ? missingValue : 0.0f, reverse, enableSkipping, HalfFloatPoint.BYTES);
```

Comment:
I assume half float support doesn't support negative values? is that right?

Reply:
No it has negative values, however we always pass missing value. I added in this new commit, it will never be null for `half_float`. I copoed this from FloatComparator, where it has the same constructor logic.